### ISSUE TITLE
[Bug] RPC Server Python Exception Can't Be Send to RPC Client

### ISSUE THREAD

**Issue by Johnson9009** [apache/tvm#16707]
After upgrade with the V0.15.0, we found the RPC have a bug, if the RPC server have some exception, we can't see the error message in RPC client like below.

![image](https://github.com/apache/tvm/assets/19802523/24ed89a5-23db-4260-8c38-79b032f763dd)

After some investigations, I found only if the exception is raise by a Python packed function will cause this issue, like the below experiment, the C++'s exception is thrown to Python side correctly, then the Python throw it again will lose the error message, the RPC client only can receive the exception but the error message in it is empty.
![image](https://github.com/apache/tvm/assets/19802523/2b982e95-7540-4bb2-b7f0-1f42ea3144d1)

![image](https://github.com/apache/tvm/assets/19802523/044599ee-a7a2-4c3b-b3c7-20c4f962f627)

If the exception is happened in a pure C++ remote packed function, then the error message is correctly sent back to the RPC client.

How to reproduce ?
just hard code a `raise "xxxxx"` in the function begin of `load_module` in `python/tvm/rpc/server.py`, or other Python packed function, and then call it in RPC client.

**Comment by Johnson9009** [comment]
![image](https://github.com/apache/tvm/assets/19802523/06df6aac-170d-4861-9ff3-b1d5bddf814c)
Is it relevant to this change?

**Comment by tqchen** [comment]
maybe indeed related to #15596 @Lunderberg seems we need to stringify python errors if they are caught by RPC

**Comment by Lunderberg** [comment]
Agreed.  The full stack trace is in the python object, so we should be able to serialize it for RPC use.  Prior to #15596, the full stack trace was embedded into the string error message, which worked with RPC transfers, but made it quite difficult to track nested levels of error messages.

I'm thinking that RPC's serialization will be the reverse of the parsing that occurs [here](https://github.com/apache/tvm/blob/main/python/tvm/_ffi/base.py#L438), so that the stack trace objects can be rebuilt when received.  (Though, missing the full python variable information available within a non-RPC `err.__traceback__`.)

**Comment by tqchen** [comment]
@Lunderberg given you introduced python trace, can you help on this one ?

**Comment by Lunderberg** [comment]
Thank you for the ping, and I probably can in a few weeks, but am currently low on available time.
